### PR TITLE
Add ability to round rectangles and add dotted borders

### PR DESCRIPTION
**Pull request by stephanie-lee**

### CODE REVIEW

File: Sources/UIImage+Effects.swift [hunk 1]
```diff
@@ -43,17 +43,73 @@ extension UIImage {
...
+     Returns a transparent rectangle of given size with a border respecting the given radius.
+
+     - Parameters:
+       - color:     The color for the image.
+       - size:      The size of the image.
+       - radius:    The radius used if drawing rounded corners. Defaults to nil.
+       - isDashed:  A boolean that states if the drawn border is solid or dashed
+     - Returns:     Returns an image with a solid color of the given size.
+    */
+    public convenience init(color: UIColor, size: CGSize, radius: CGFloat = 0, isDashed: Bool) throws {
```
Comment (stephanie-lee) [inline Sources/UIImage+Effects.swift]: Not sure this is appropriate as a convenience init, but I am thinking that the border is bound to the drawn rectangle and not whatever image I will be `union`-ing it with.

File: Sources/UIImage+Effects.swift [hunk 2]
```diff
@@ -43,17 +43,73 @@ extension UIImage {
...
+            throw ImageError.CantCreateImage
+        }
+        self.init(cgImage: cgimage, scale: scale, orientation: .up)
+    }
+
+    /**
+     Returns a transparent rectangle of given size with a border respecting the given radius.
+
+     - Parameters:
+       - color:     The color for the image.
```
Comment (eb-smith-affirm) [inline Sources/UIImage+Effects.swift]: Color for the border.

File: Sources/UIImage+Effects.swift [hunk 3]
```diff
@@ -43,17 +43,73 @@ extension UIImage {
...
+       - size:      The size of the image.
+       - radius:    The radius used if drawing rounded corners. Defaults to nil.
```
Comment (eb-smith-affirm) [inline Sources/UIImage+Effects.swift]: Defaults to zero.

File: Sources/UIImage+Effects.swift (hunk 1 above)
Comment (eb-smith-affirm) [inline Sources/UIImage+Effects.swift]: Set a default value for `isDashed`?

File: Sources/ViewController.swift [hunk 4]
```diff
@@ -107,4 +109,12 @@ class ViewController: UIViewController {
         )
         return labelImage
     }
+
+    private func iconWithRoundedRectangle(name: String) -> UIView {
+        let icon = AFIcon.named(name, scheme: .grayscale, size: CGSize(width: 50, height: 50))
+        let image = try? UIImage(color: .yellow, size: CGSize(width: 200, height: 92), radius: 50, isDashed: true)
+
+        let newImage = try? image!.union(icon)
```
Comment (eb-smith-affirm) [inline Sources/ViewController.swift]: Don't force unwrap here, it's possible to crash (though unlikely).

**Review by eb-smith-affirm — CHANGES_REQUESTED** [review]
Nice additions - These will be useful.

File: Sources/UIImage+Effects.swift (hunk 1 above)
Comment (stephanie-lee) [reply Sources/UIImage+Effects.swift]: Yea sounds good, set to false

File: Sources/ViewController.swift (hunk 4 above)
Comment (stephanie-lee) [reply Sources/ViewController.swift]: Curious what the correct implementation would be then? Technicallyyyyyy, we force unwrap when we try to put it into the labeled image. 

In Affirm production, I'm doing `modifiedImage ?? originalImage`. Does that seem like the correct way to fail?

Comment (stephanie-lee) [reply Sources/ViewController.swift]: https://github.com/E-B-Smith/Colors/pull/5/commits/99e51bffade1d9ff7514f7d03e902cf5ac834524

Comment (eb-smith-affirm) [reply Sources/ViewController.swift]: That seems like a good way to fail. 

In practice I use a default dummy image as a place holder that I expect will always work unless something is really messed up. Like:
```
    private func iconWithRoundedRectangle(name: String) -> UIView {
        var image = UIImage(named: "cry")! // I expect this to always be available, for instance.
        if let newImage = try?
            UIImage(color: .red, size: CGSize(width: 200, height: 92), radius: 20, isDashed: true)
                .union(AFIcon.named(name, scheme: .grayscale, size: CGSize(width: 50, height: 50))) {
            image = newImage
        }
        return LabeledImage(text: "Icon with oval background", image: image)
    }
```

**Review by E-B-Smith — APPROVED** [review]

**Pull request merged**